core/src/com/unciv/ui/tutorials/TutorialController.kt

```diff
@@ -70,10 +71,20 @@ class TutorialController(screen: BaseScreen) {
     /** Get all Tutorials intended to be displayed in the Civilopedia
      *  as a List of wrappers supporting INamed and ICivilopediaText
      */
-    fun getCivilopediaTutorials() =
-        tutorials.filter {
+    fun getCivilopediaTutorials(ruleset: Ruleset): List<CivilopediaTutorial> {
+        val civilopediaTutorials = tutorials.filter {
             Tutorial.findByName(it.key)!!.isCivilopedia
-        }.map {
-            CivilopediaTutorial(it.key, it.value)
+        }.map { tutorial ->
+            val lines = tutorial.value
+                
+            if (tutorial.key == "Unhappiness") {
+                for (unhappinessEffect in ruleset.unhappinessEffects.values.sortedByDescending { it.unhappiness }) {
+                    lines.add("\n${unhappinessEffect.toCivilopediaLines()}")
+                }
+            }
+            
+            CivilopediaTutorial(tutorial.key, lines)
```

Comment:
Nicely done.
Wasn't there another virtually generated page in there, to compare....? No, that was Religion, wasn't it?
And - "\n": Is that a FormattedLine collection or Strings? If the former, there is a "styled" empty line with a global-const controlled spacing (smaller than an entire line when I designed it) - if I remember correctly - thus mixing gives barely detectable line spacing inconsistencies.

Reply:
> Nicely done. Wasn't there another virtually generated page in there, to compare....? No, that was Religion, wasn't it? 

I can't think of one, and the quick look I cast on the civilopedia code before writing this didn't reveal any other generated pages like this.

>And - "\n": Is that a FormattedLine collection or Strings? If the former, there is a "styled" empty line with a global-const controlled spacing (smaller than an entire line when I designed it) - if I remember correctly - thus mixing gives barely detectable line spacing inconsistencies.

I have no idea, to be honest, but the line breaks looked fine, and especially if its 'barely detectable', I don't think anyone will notice or mind it.

Reply:
OK, `lines` in there is strings, a compromise back then, so `\n` is fine